distributions/openhab/src/main/resources/bin/update.ps1

```diff
@@ -69,190 +12,582 @@ Function Update-openHAB {
         }
+    }
+    catch {
+        return PrintAndReturn "Could not stop/delete the openHAB windows server - please do that manually and try again" $_
+    }
 
+    Write-Host -ForegroundColor Cyan "Changing location to $OHDirectory"
+    try {
+        Set-Location -Path $OHDirectory
+    }
+    catch {
+        return PrintAndReturn "Could not change location to $OHDirectory - exiting" $_
+    }
 
-        # Checking if openHAB is running
-        Write-Host -ForegroundColor Cyan "Checking whether openHAB is running..."
-        $m = (Get-WmiObject Win32_Process -Filter "name = 'java.exe'" |
-              where { $_.CommandLine.Contains("openhab") } | measure)
-        if ($m.Count -gt 0) {
-            throw "openHAB seems to be running, stop it before running this update script"
-        }
-
+    # Download the selected openHAB version
+    # Choose bintray for releases, jenkins for snapshots.
+    $TempDir = "$(GetOpenHABTempDirectory)\update"
+    $OutputFile = "$TempDir\openhab-$OHVersion.zip"
+    $TempDistributionZip = "$TempDir\openhab-$OHVersion.zip";
+    $TempDistribution = "$TempDir\openhab-$OHVersion"
+    if ($Snapshot) {
+        $OHVersion = "$OHVersion-SNAPSHOT"
+        $DownloadLocation="https://ci.openhab.org/job/openHAB-Distribution/lastSuccessfulBuild/artifact/distributions/openhab/target/openhab-$OHVersion.zip"
+        $AddonsDownloadLocation="https://ci.openhab.org/job/openHAB-Distribution/lastSuccessfulBuild/artifact/distributions/openhab-addons/target/openhab-addons-$OHVersion.kar"
+        $LegacyAddonsDownloadLocation="https://ci.openhab.org/job/openHAB-Distribution/lastSuccessfulBuild/artifact/distributions/openhab-addons-legacy/target/openhab-addons-legacy-$OHVersion.kar"
+    }
+    else {
+        $DownloadLocation = "https://bintray.com/openhab/mvn/download_file?file_path=org%2Fopenhab%2Fdistro%2Fopenhab%2F$OHVersion%2Fopenhab-$OHVersion.zip"
+        $AddonsDownloadLocation = "https://bintray.com/openhab/mvn/download_file?file_path=org%2Fopenhab%2Fdistro%2Fopenhab-addons%2F$OHVersion%2Fopenhab-addons-$OHVersion.kar"
+        $LegacyAddonsDownloadLocation = "https://bintray.com/openhab/mvn/download_file?file_path=org%2Fopenhab%2Fdistro%2Fopenhab-addons-legacy%2F$OHVersion%2Fopenhab-addons-legacy-$OHVersion.kar"
+    }
 
-        # Backup openHAB only if not coming via new update script
-        if ($SkipNew -eq $false)
-        {
-            Write-Host ""
-            Write-Host -ForegroundColor Magenta "Backup script starting..."
-            Write-Host -ForegroundColor Cyan "Making a backup in '$OHDirectory\backups' ..."
-            $BackupScript = Join-Path (pwd) '\runtime\bin\backup.ps1'
-            . $BackupScript
-            Backup-openHAB -OHDirectory (pwd)
-            Write-Host -ForegroundColor Magenta "Backup script finished."
-            Write-Host ""
+    if ($Snapshot) {
+        Write-Host -ForegroundColor Yellow "The current version is $CurrentVersion-SNAPSHOT"
```

Comment:
This doesn't make any sense. Just because I choose to upgrade to a snapshot, doesn't mean I have a snapshot.